### DISCUSSION TITLE
Errors appear when using 'make -j4'

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#18574] [Q&A Getting Started] [answered]
Hi Moose expert or user,

I am studying how to create a Moose application following this page (https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/step01_moose_app.html). There are some errors appearing when I try 'make -j4' in the step of Compile the Application Executable.  Could anyone help me fix this issue? Thank you.
[errors.txt](https://github.com/idaholab/moose/files/6963675/errors.txt)

**Comment by GiudGiud** [comment]
Hello

Are you using a mac?
If so look at the bottom of this page for the relevant fix (something about hostname)
https://mooseframework.inl.gov/help/troubleshooting.html

Guillaume

**Reply by xchengood** [reply]
> Hello
> 
> Are you using a mac?
> If so look at the bottom of this page for the relevant fix (something about hostname)
> https://mooseframework.inl.gov/help/troubleshooting.html
> 
> Guillaume

Hi @GiudGiud. Yes, it is a mac. It doesn't work and still shows the same errors after trying the hostname solution according to suggestions.

**Reply by GiudGiud** [reply]
could you please paste the errors here

**Reply by xchengood** [reply]
> could you please paste the errors here

[errors.txt](https://github.com/idaholab/moose/files/6963895/errors.txt)

**Reply by GiudGiud** [reply]
so it compiles with `make` but not with `make -j 4`?
Did you update anything?
This seems like xcode got updated, we dont support the latest version

**Reply by xchengood** [reply]
> so it compiles with `make` but not with `make -j 4`?
> Did you update anything?
> This seems like xcode got updated, we dont support the latest version

It shows the same errors when only using 'make'.
I updated the system to Macox Big Sur 2 days ago. I noticed there will appear some errors with the newest Xcode. So I downgrade its version to 12.4 according to instructions in another post.
I have no idea why those errors above occur.

**Reply by GiudGiud** [reply]
so `/usr/bin/xcodebuild -version` returns 12.4?

**Reply by xchengood** [reply]
> /usr/bin/xcodebuild -version

It does not. I got the following error. But there is indeed an Xcode with 12.4 in my mac.
<img width="565" alt="Screen Shot 2021-08-10 at 8 24 12 PM" src="https://user-images.githubusercontent.com/74226455/128955624-763a8c29-bbf7-40a7-8aad-93c39849f26f.png">

**Reply by GiudGiud** [reply]
Ah ok let's get the version of xcode CLT to double check
Maybe try 
`xcode-select --version`
and 
`pkgutil --pkg-info=com.apple.pkg.CLTools_Executables` and look under version as Daniel suggested

**Reply by xchengood** [reply]
> Ah ok let's get the version of xcode CLT to double check
> Maybe try
> `xcode-select --version`
> and
> `pkgutil --pkg-info=com.apple.pkg.CLTools_Executables` and look under version as Daniel suggested

Thank you. The issue has been resolved according to your and Daniel's suggestion. I appreciate it.

**Comment by dschwen** [comment]
Did you maybe install (or accidentally upgrade to) a version of XCode CMD line tools other than 12.4?

**Reply by dschwen** [reply]
Actually that is _most definitely_ what happend to you, as it just happened to me as well. My mac decided to force that update on me 👀  and I'm seeing the exact same errors. Type

```
pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
```

I'm pretty sure you'll see the same thing under `version`:

```
package-id: com.apple.pkg.CLTools_Executables
version: 12.5.1.0.1.1623191612
volume: /
location: /
install-time: 1628330871
groups: com.apple.FindSystemFiles.pkg-group
```

**Reply by dschwen** [reply]
Confirm your XCode CMD Line Tools path

```
(moose) fn606041:test schwd$ xcode-select -p
/Library/Developer/CommandLineTools
```

Delete the whole directory:
```
(moose) fn606041:test schwd$ sudo rm -rf /Library/Developer/CommandLineTools
Password:
```

Download version 12.4 (_Command Line Tools for Xcode 12.4_) from developer.apple.com and install it. Bam, fixed!

**Reply by xchengood** [reply]
> Confirm your XCode CMD Line Tools path
> 
> ```
> (moose) fn606041:test schwd$ xcode-select -p
> /Library/Developer/CommandLineTools
> ```
> 
> Delete the whole directory:
> 
> ```
> (moose) fn606041:test schwd$ sudo rm -rf /Library/Developer/CommandLineTools
> Password:
> ```
> 
> Download version 12.4 (_Command Line Tools for Xcode 12.4_) from developer.apple.com and install it. Bam, fixed!

Thank you for your suggestion. The issue has been resolved well. Appreciate.

**Comment by cticenhour** [comment, marked as answer]
For those who experience a similar issue in the future, I am re-answering this question now that MOOSE also supports Xcode 12.5, 13, and 13.1. Please update your conda packages to the newest versions by performing 
```
conda update --all
```
within your MOOSE conda environment. See #19289 for more information.
